modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1516,6 +1517,10 @@ private void _validateListTypeEntryValue(
 			}
 
 			if (objectField.getListTypeDefinitionId() == 0) {
+				if (StringUtil.equals(objectField.getType(), "String")) {
```

Comment:
Hey @mateussgn , this method is responsible for ListTypeEntry value validation, would be better to call your new method directly inside addObjectEntry, as it does another kind of validation.

Maybe you did this way to avoid loop over map entries again, if you want this, it would be nice to make this method more generic, renaming this method to _validate or _validateValues and making the code more clear with something like: 

if (objectField.getListTypeDefinitionId() != 0) {
_validateListTypeEntries()
} 
else {
_validateObjectFieldStringTypeLength()
}